iina/MainMenuActions.swift

```diff
@@ -45,6 +45,11 @@ class MainMenuActionHandler: NSResponder {
     }
   }
 
+  @objc func menuShowCurrentFileInFinder(_ sender: NSMenuItem) {
+    guard let url = player.info.currentURL else { return }
+    NSWorkspace.shared.activateFileViewerSelecting([url])
```

Comment:
Missing check for `isNetworkResource`. Not sure how `activateFileViewerSelecting` will react if given a URL to a video on a web site. Have a look at the code in `PlaylistViewController` that calls `activateFileViewerSelecting`.

Reply:
The approach taken in the `PlaylistViewController` is to simply ignore such items, and this is indeed necessary to avoid an error when using the keyboard shortcut. I applied it. However, in our case it would have been better to disable the button in advance (and display it as gray). Is it possible?

Reply:
I noticed that the similar item, "Delete current file", throws an error when trying to delete a network file:
![CleanShot 2024-06-21 at 07 53 04@2x](https://github.com/iina/iina/assets/78599753/8f15b23e-6b94-4d5d-bb33-abfef2af616f)
We will have to apply the check there as well.

Reply:
> In our case it would have been better to disable the button in advance (and display it as gray). Is it possible?

@uiryuu?

Reply:
I implemented this using `validateMenuItem`, similar to `PlaylistViewController` and `HistoryWindowController`.

We will have to handle `menuDeleteCurrentFile` in the same way (add an internal `guard` to prevent an error in using the keyboard shortcut, and add it to `validateMenuItem` to disable the menu item). But I didn't know if it was appropriate to include it in the current PR, let me know whether to add it as well.

Reply:
Lets treat the problem with deleting as a separate issue. I will enter one for it.

Reply:
I entered #5011 for the problem with `Delete Current File` being enabled while streaming.